plugins/out_opentelemetry/opentelemetry.c

```diff
@@ -934,6 +944,186 @@ static int flush_to_otel(struct opentelemetry_context *ctx,
+            is_valid_severity_number(ra_val->o.via.u64) == FLB_TRUE) {
+            log_record->severity_number = ra_val->o.via.u64;
+            flb_ra_key_value_destroy(ra_val);
+        }
+    }
+
+    /* TraceFlags */
+    if (ctx->ra_trace_flags_metadata) {
+        ra_val = flb_ra_get_value_object(ctx->ra_trace_flags_metadata, *event->metadata);
+        if (ra_val != NULL && ra_val->o.type == MSGPACK_OBJECT_POSITIVE_INTEGER) {
+            log_record->flags = (uint32_t)ra_val->o.via.u64;
+            flb_ra_key_value_destroy(ra_val);
+        }
+    }
+
+    /* SpanId */
+    if (ctx->ra_span_id_metadata) {
+        ra_val = flb_ra_get_value_object(ctx->ra_span_id_metadata, *event->metadata);
+        if (ra_val != NULL && ra_val->o.type == MSGPACK_OBJECT_BIN) {
+            log_record->span_id.data = flb_calloc(1, ra_val->o.via.bin.size);
+            if (log_record->span_id.data) {
+                memcpy(log_record->span_id.data, ra_val->o.via.bin.ptr, ra_val->o.via.bin.size);
+                log_record->span_id.len = ra_val->o.via.bin.size;
+            }
+            flb_ra_key_value_destroy(ra_val);
+        }
+    }
+
+    /* TraceId */
+    if (ctx->ra_trace_id_metadata) {
+        ra_val = flb_ra_get_value_object(ctx->ra_trace_id_metadata, *event->metadata);
+        if (ra_val != NULL && ra_val->o.type == MSGPACK_OBJECT_BIN) {
+            log_record->trace_id.data = flb_calloc(1, ra_val->o.via.bin.size);
+            if (log_record->trace_id.data) {
+                memcpy(log_record->trace_id.data, ra_val->o.via.bin.ptr, ra_val->o.via.bin.size);
+                log_record->trace_id.len = ra_val->o.via.bin.size;
+            }
+            flb_ra_key_value_destroy(ra_val);
+        }
+    }
+
+    /* Attributes */
+    if (ctx->ra_attributes_metadata) {
+        ra_val = flb_ra_get_value_object(ctx->ra_attributes_metadata, *event->metadata);
+        if (ra_val != NULL && ra_val->o.type == MSGPACK_OBJECT_MAP) {
+            if (log_record->attributes != NULL) {
+                otlp_kvarray_destroy(log_record->attributes,
+                                     log_record->n_attributes);
+            }
+            log_record->attributes = msgpack_map_to_otlp_kvarray(&ra_val->o, &log_record->n_attributes);
+            flb_ra_key_value_destroy(ra_val);
+        }
+    }
+
+    return 0;
+}
+
+static int append_v1_logs_message(struct opentelemetry_context *ctx,
+                                   struct flb_log_event *event,
+                                   Opentelemetry__Proto__Logs__V1__LogRecord  *log_record)
+{
+    struct flb_ra_value *ra_val;
+
+    flb_plg_info(ctx->ins, "Appending logs for message");
+
+    if (ctx == NULL || event == NULL || log_record == NULL) {
+        return -1;
+    }
+
+    /* SpanId */
+    if (ctx->ra_span_id_message) {
+        flb_plg_info(ctx->ins, "span id message not null");
+        ra_val = flb_ra_get_value_object(ctx->ra_span_id_message, *event->body);
```

Comment:
@nokute78 I have piggy backed off your changes.  However my solution does not seem to be working when I test.  Based on my test, the ra_val seems be null or not of type msdpack object bin.  The if statement below is evaluating as false.  Please take a look and any suggestions would be greatly appreciated.